src/utils/Material.js

```diff
@@ -82,13 +82,39 @@ material.loadTextureAsync = function (src) {
 };
 
 material.loadVideoTextureAsync = function (video) {
+    video.src = html.resolvePath(video.src);
+
+    video.muted = true; // Mute the video to allow autoplay
+    video.autoplay = false; //true; // Attempt to autoplay
+
     return new Promise((resolve, reject) => {
-        try {
-            const textureLoader = new THREE.VideoTexture(video);
-            resolve(textureLoader);
-        } catch(err) {
-            reject(err);
-        }
+        // Event listener to ensure video is ready
+        video.onloadeddata = () => {
+            try {
+                const textureLoader = new THREE.VideoTexture(video);
```

Comment:
Nit for an existing issue: Isn't the name `textureLoader` for `VideoTexture` instance confusing because Three.js has `TextureLoader` class. Maybe `texture` or `videoTexture` would be clearer?

Reply:
√